From 923849fa1cc36580086be542036c4935d7cbfe1c Mon Sep 17 00:00:00 2001
From: Petr Lautrbach <plautrba@redhat.com>
Date: Mon, 23 May 2022 12:25:58 +0200
Subject: [PATCH] libsepol: Bring back POLICYDB_CAPABILITY_* constants
Content-type: text/plain

They were removed by the commit 0d84ebcbc475d ("libsepol: Shorten the
policy capability enum names") but they might be used somewhere in code
deployed on RHEL 9.0

Signed-off-by: Petr Lautrbach <plautrba@redhat.com>
---
 libsepol/include/sepol/policydb/polcaps.h | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/libsepol/include/sepol/policydb/polcaps.h b/libsepol/include/sepol/policydb/polcaps.h
index f5e32e60975d..e568733b3a33 100644
--- a/libsepol/include/sepol/policydb/polcaps.h
+++ b/libsepol/include/sepol/policydb/polcaps.h
@@ -19,6 +19,20 @@ enum {
 };
 #define POLICYDB_CAP_MAX (__POLICYDB_CAP_MAX - 1)
 
+/* legacy constants from past to preserve backward compatibility */
+enum {
+    POLICYDB_CAPABILITY_NETPEER =                 POLICYDB_CAP_NETPEER,
+    POLICYDB_CAPABILITY_OPENPERM =                POLICYDB_CAP_OPENPERM,
+    POLICYDB_CAPABILITY_EXTSOCKCLASS =            POLICYDB_CAP_EXTSOCKCLASS,
+    POLICYDB_CAPABILITY_ALWAYSNETWORK =           POLICYDB_CAP_ALWAYSNETWORK,
+    POLICYDB_CAPABILITY_CGROUPSECLABEL =          POLICYDB_CAP_CGROUPSECLABEL,
+    POLICYDB_CAPABILITY_NNP_NOSUID_TRANSITION =   POLICYDB_CAP_NNP_NOSUID_TRANSITION,
+    POLICYDB_CAPABILITY_GENFS_SECLABEL_SYMLINKS = POLICYDB_CAP_GENFS_SECLABEL_SYMLINKS,
+    POLICYDB_CAPABILITY_IOCTL_SKIP_CLOEXEC =      POLICYDB_CAP_IOCTL_SKIP_CLOEXEC,
+    __POLICYDB_CAPABILITY_MAX =                 __POLICYDB_CAP_MAX
+};
+#define POLICYDB_CAPABILITY_MAX (__POLICYDB_CAP_MAX - 1)
+
 /* Convert a capability name to number. */
 extern int sepol_polcap_getnum(const char *name);
 
-- 
2.39.0

